### ISSUE TITLE
Web - Unresponsive 'request a code here' link on 'Magic code expired' page

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21351]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to sign-in page
2. Enter email > click on continue button
3. Wait for request new code timeout
4. Click on 'Resend magic code' link
5. Open email > copy the expired link
6. Open new tab > paste the link > prefix the link with staging> press enter
7. Click on 'request a code here' link

## Expected Result:
The link should be responsive if it is not disabled or,
* The link should be disabled if it is intended to make it unresponsive

## Actual Result:
Unresponsive 'request a code here'  link

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.31.1

**Reproducible in staging?:** yes

**Reproducible in production?:** new feature

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/58206bd3-81f9-44da-8f21-7370a6738e88

https://github.com/Expensify/App/assets/93399543/cfd820d7-50ce-4354-823c-49d4746d64a0

**Expensify/Expensify Issue URL:**

**Issue reported by:** @Natnael-guchima

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687459746529029

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by yuwenmemon** [comment]
@cristipaval or @NikkiWines seems like this is a side-effect of the throttling we're doing for the `requestValidateCode`. I see that we do make a request, but we don't send the code:

https://github.com/Expensify/App/assets/4741899/e973f437-26ae-40cf-b07c-9773a419ad9f

**Comment by NikkiWines** [comment]
Hmm, definitely we can fix this, though not sure it needs to be a blocker. I wonder, since we're migrating to [timers](https://github.com/Expensify/App/pull/20732) for the resend validate code button in the main sign in flow if we should just direct the user back to the sign in page instead of allowing them to request a code from that view 🤔 

What do you think, @cristipaval? cc'ing @johnmlee101 as well since you have context

**Comment by cristipaval** [comment]
yes @yuwenmemon you're right, this is a side-effect of the throttling.

@NikkiWines Yes I think we can remove the blocker label. I can work on a fix for this.
In regards to the removal of the link in that view, I would love to hear what @shawnborton thinks

**Comment by shawnborton** [comment]
Could we just change out that whole line that says "You can request a new code here" to something like "Too many requests received. Please wait a moment" and then flip it back when the timer is up? I don't quite like the idea of an actual countdown timer.

**Comment by cristipaval** [comment]
Ok, so instead of removing the link to resend the code, you still want to keep it in the **Magic code expired** page, and behave as you just described, correct @shawnborton ?

**Comment by shawnborton** [comment]
Yeah, that feels better than just sending them back to the sign in page with no context or message.

**Comment by NikkiWines** [comment]
Ah, I was thinking we could keep the expired page but instead of allowing them to click a link to request a code, we'd provide a link that directs them back to the sign-in page where they'd go through the sign-in process to get a new code.

**Comment by shawnborton** [comment]
Hmm I feel like that might be worse UX. Rather than making them sign in again, can't we just wait and then resurface the link when it's ready?

**Comment by NikkiWines** [comment]
Hmm, I'm not sure I understand @shawnborton, sorry. As I understand, this is the logic surrounding the login flow when the user has an expired code. They'll have two tabs: 

1. Tab A, open to the `Please enter the magic code sent to your login` view (which also has a the `Didn't receive a magic code` link to request a new code)
![image](https://github.com/Expensify/App/assets/16747822/e80e2b04-8212-40a9-9bc3-3ac1c390b15b)

2. Tab B, open to the expired magic code view(e.g. new.expensify.com/v/123/345678) 
![Screenshot 2023-06-23 at 11 26 23](https://github.com/Expensify/App/assets/16747822/734ddc46-878b-4825-82b3-9779002182d7)

If they don't have tab A, we won't show the expired magic code view, because there's nowhere that authenticates the code to tell us it's invalid. Without tab A, they also can't log in because there's no window listening for the code and no place to manually enter the code.

Giving them a place in tab B to request a code just means we have to maintain error messaging in both views and tab B is redundant either way because it won't ever show the updated code. To me, it makes sense to move them away from that view by directing them to request a new code from the sign-in page (in this case, tab A, or an entirely new tab).

**Comment by shawnborton** [comment]
Ah okay, I think I am understanding better now. Okay, so that being said, let's go with your recommendation then.

**Comment by cristipaval** [comment]
Opening a PR soon.

**Comment by cristipaval** [comment]
Well, Melvin, this was deployed to production yesterday

**Comment by cristipaval** [comment]
@stephanieelliott could you please help with the payment for the C+ review?

**Comment by stephanieelliott** [comment]
Extended the offer to @Natnael-Guchima on Upwork for the reporting bonus! Holding on C+ payment.

**Comment by stephanieelliott** [comment]
Still holding on @parasharrajat's payment. for newdot setup

**Comment by stephanieelliott** [comment]
Summarizing payouts for this issue: 

 - Issue reporter: $250 @Natnael-Guchima (payment via Upwork)
 - Contributor: N/A
 - Contributor+: $1000 @parasharrajat (payment requested via newdot)
 
 Upwork job is [here](https://www.upwork.com/ab/applicants/1680824738964025344/job-details) ,no bonus as this was fixed by an internal engineer.

**Comment by JmillsExpensify** [comment]
Reviewed details for @parasharrajat. These details are accurate based on summary from Business Reviewer and are now approved for payment in NewDot.
